src/Weaviate.Client.Tests/Integration/TestRbacRoles.cs

```diff
@@ -0,0 +1,301 @@
+namespace Weaviate.Client.Tests.Integration;
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Weaviate.Client.Models;
+using Xunit;
+
+[Trait("Category", "RBAC")]
+public class TestRbacRoles : IntegrationTests
+{
+    public override ushort RestPort => 8092;
+    public override ushort GrpcPort => 50063;
+    private const string ADMIN_API_KEY = "admin-key";
+    public override ICredentials? Credentials => Auth.ApiKey(ADMIN_API_KEY);
+
+    private static string MakeRoleName(string suffix) =>
+        $"test-role-{suffix}-{Random.Shared.Next(1, 1_000_000)}";
+
+    public TestRbacRoles()
+    {
+        RequireVersion("1.30.0");
+    }
+
+    [Fact]
+    public async Task ListRoles()
+    {
+        RequireVersion("1.30.0");
+        var roles = (await _weaviate.Roles.ListAll()).ToList();
+        Assert.NotEmpty(roles);
+        Assert.Contains(roles, r => r.Name == "viewer");
+    }
+
+    [Fact]
+    public async Task CreateRoleWithPermissions()
+    {
+        RequireVersion("1.30.0");
+        var roleName = MakeRoleName("create");
+        try
+        {
+            await _weaviate.Roles.Delete(roleName);
+
+            var created = await _weaviate.Roles.Create(
+                roleName,
+                [new Permissions.Roles(new RolesResource(roleName, null)) { Read = true }]
```

Comment:
very nice syntax!

Reply:
This one has a leftover. It can also be written as:
`new Permissions.Roles(roleName) { Read = true }` or `new Permissions.Roles(new (roleName)) { Read = true }`